lib/testobject.js

```diff
@@ -207,11 +212,25 @@ TestObject.fetchAppium = async function (driverName, driverUrl) {
     proc.start();
   });
   // Zip it and return it
+  const pathToAppiumCopy = path.resolve(tempdir, 'appium');
   const pathToZip = path.resolve(tempdir, 'appium.zip');
-  logger.debug(`Writing zip file to ${pathToZip}`);
-  await exec('zip', ['-r', 'appium.zip', 'appium/'], {cwd: tempdir});
-  await fs.rimraf(path.resolve(tempdir, 'appium'));
+
+  //await zip(pathToZip, [pathToAppiumCopy]);
+  const pathToBestZip = path.resolve(process.env.PWD, 'node_modules', '.bin', 'bestzip');
+  logger.debug(`Running from ${pathToBestZip}`);
+  await exec(pathToBestZip, [pathToZip, pathToAppiumCopy]);
+  logger.debug(`Zipped Appium to ${pathToZip}`);
+  await fs.rimraf(pathToAppiumCopy);
```

Comment:
shouldn't this be in `finally` block?

Reply:
All these years of Javascript development and I'm just learning now that Javascript has a `finally` clause

Reply:
I don't understand why it should be in a `finally` block. There's no try-catch around it.

Reply:
because if some exception happens in between the temporary file is created and removed then it will stay and use the space